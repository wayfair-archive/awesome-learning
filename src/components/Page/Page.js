import React, { useRef, useEffect } from 'react';
import './Page.scss';

const Page = ({ title, children }) => {
  const pageRef = useRef();

  useEffect(() => {
    pageRef.current.scrollIntoView();
  });

  return (
    <div ref={pageRef} className="Page">
      <div className="Page-inner">
        { title && <h1 className="Page-title">{title}</h1>}
        <div className="Page-body">
          {children}
        </div>
      </div>
    </div>
  );
};

export default Page;