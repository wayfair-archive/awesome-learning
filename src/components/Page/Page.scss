.Page {

  &-inner {
    padding: 25px 0;
    @include breakpoint-sm {
      padding: 30px 0;
    }

    @include breakpoint-md {
      padding: 40px 0;
    }

    @include breakpoint-lg {
      padding: 40px 35px;
    }
  }

  &-title {
    font-size: $font--3500;
    font-weight: $font-weight--bold;
    @include line-height(2);
    @include margin-top(0);
    @include margin-bottom(1.45);
  }

  &-body {
    @include line-height(1);
    @include margin(0, 0, 0);
  }
}
