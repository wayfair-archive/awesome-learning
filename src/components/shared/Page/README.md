The `Page` is a compositee component that can be used to build up a page. It consists of a `title` and `children` which can be
any number of nodes that build up the content of the page.

## Props

The following props may be passed to configure the Block.

| name         | type                 | description                                            | default |
| ------------ | -------------------- | ------------------------------------------------------ | ------- |
| **title**    | `String`             | The title of the section                               | `null`  |
| **children** | `node(s) - required` | The child components that make up the rest of the page |         |
