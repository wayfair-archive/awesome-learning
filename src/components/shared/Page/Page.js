import React from 'react';
import PropTypes from 'prop-types';
import './Page.scss';

const Page = ({children}) => (
  <div className="Page">
    <div className="Page-inner">
      <div className="Page-body">{children}</div>
    </div>
  </div>
);

Page.propTypes = {
  children: PropTypes.node,
};

Page.defaultProps = {
  children: null,
};

export default Page;
