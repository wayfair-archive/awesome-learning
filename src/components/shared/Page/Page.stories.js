import React from 'react';
import {storiesOf} from '@storybook/react';
import {withKnobs, text} from '@storybook/addon-knobs';
import Page from './Page';
import PageReadme from './README.md';

const pageText = `
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  Nulla at volutpat diam ut venenatis tellus in metus vulputate. Sodales neque sodales ut etiam.
  Diam phasellus vestibulum lorem sed risus ultricies tristique nulla. In cursus turpis massa tincidunt dui ut ornare lectus sit.`;

const stories = storiesOf('Shared.Page', module);

stories
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      // Show readme at the addons panel
      sidebar: PageReadme,
    },
  })
  .add('basic', () => (
    <Page title={text('title', 'Page Title')}>{pageText}</Page>
  ));
