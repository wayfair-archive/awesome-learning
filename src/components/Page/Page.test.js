import React from 'react';
import renderer from 'react-test-renderer';
import Page from './Page';

describe('Page', () => {
  const props = {
    children: 'test',
    title: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Page {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
