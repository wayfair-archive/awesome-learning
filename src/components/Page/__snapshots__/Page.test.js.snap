// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div
  className="Page"
>
  <div
    className="Page-inner"
  >
    <h1
      className="Page-title"
    >
      test
    </h1>
    <div
      className="Page-body"
    >
      test
    </div>
  </div>
</div>
`;
